components/kubelet/pod/component.go

```diff
@@ -152,48 +175,34 @@ type Data struct {
 	ts time.Time `json:"-"`
 	// error from the last check
 	err error `json:"-"`
-	// set to true if the error is the connection error to kubelet
-	connErr bool `json:"-"`
 }
 
-func (d *Data) getReason() string {
+func (d *Data) getReason(failedCount int) string {
 	if d == nil || (d.err == nil && len(d.Pods) == 0) {
 		return "no pod found or kubelet is not running"
 	}
 
 	if d.err != nil {
-		if d.connErr {
+		if isConnectionRefusedError(d.err) {
 			// e.g.,
 			// Get "http://localhost:10255/pods": dial tcp [::1]:10255: connect: connection refused
 			return fmt.Sprintf("connection error to node %q -- %v", d.NodeName, d.err)
 		}
 
-		return fmt.Sprintf("failed to list pods from kubelet read-only port -- %v", d.err)
+		return fmt.Sprintf("list pods from kubelet read-only port failed %d time(s) (%v)", failedCount, d.err)
```

Comment:
1. we should generate the reason in the checkOnce before setting up lastData.
2. we should not generate the reason with d.err since d.err will be passed to State.Error which will be showed as MoreInformation in alert..